stream-tooling.md

```diff
@@ -0,0 +1,86 @@
+# Stream tooling
+
+## Introduction
+
+FCOS will have multiple streams:
+
+| Type | Name (SCM branch and ostree ref) | ostree repo |
+| -- | -- | -- |
+| Production | next | prod |
+| Production | testing | prod |
+| Production | stable | prod |
+| Development | testing-devel | annex |
+| Development | next-devel | annex |
+| Mechanical | rawhide | annex |
+| Mechanical | branched | annex |
+| Mechanical | bodhi-updates | annex |
+| Mechanical | bodhi-updates-testing | annex |
+
+Development and mechanical streams are subject to change.
+
+We need a way to both (1) fix the content set for a particular stream release, and (2) integrate new content into development streams.
+
+## Current tools at our disposal
+- git
+- rpm-ostree treefiles: manifest fed to rpm-ostree that contains the list of packages to use during a compose. [Example](https://github.com/coreos/fedora-coreos-config/blob/master/fedora-coreos-base.yaml).
+- rpm-ostree treefile locks: [pending rpm-ostree patch]( https://github.com/projectatomic/rpm-ostree/pull/1745) adding "lockfile" functionality similar to Cargo.lock/Gopkg.lock. This essentially means that the rpm-ostree compose is guaranteed to use specific package versions (or fail) as described in the lockfile. (To be clear, all of the below could probably be done without a lock file, since the treefile supports fully specifying the NEVRA, but having a separate lockfile allows for more sophisticated tooling and a cleaner treefile.)
+- Koji tags: a way to track packages built in Koji. Koji is capable of creating yum repos from such tags. RPM builds may be "tagged" in so that the next repo regeneration includes it.
+- [dist-git](http://src.fedoraproject.org/): git where RPM spec files are kept and Koji builds source from.
+
+## Proposal
+
+**Mechanical** streams are not curated; they’re automated nightly snapshots of the underlying repos.  They source their RPMs from the regular Fedora repos (using 30 here to mean `$currentrelease`):
+1. **rawhide** <- f32
+2. **branched** <- f31 when a branch exists, otherwise tracks **rawhide**
+3. **bodhi-updates** <- f30-stable + f30-updates
+4. **bodhi-updates-testing** <- f30-stable + f30-updates + f30-updates-testing
+
+**Production** streams are intended for production use.  They source their RPMs from a _single_ Koji tag, `coreos-release`, from which we create a yum repo:
+1. **next** <- coreos-release
+2. **testing** <- coreos-release
+3. **stable** <- coreos-release
```

Comment:
This might be an implementation detail we are intending to gloss over here but my understanding is that we would still use the `coreos-pool` related dist-repo (yum repo) for the inputs for all composes (the lockfiles control the actual packages that get used so it doesn't matter as long as the rpms are available) and we would tag the packages into the `coreos-release` tag once they had made it into a release.

Reply:
Yup, that was my understanding as well.  Updated.